alica_engine/src/engine/AlicaContext.cpp

```diff
@@ -109,10 +109,17 @@ void AlicaContext::stepEngine()
     _engine->stepNotify();
     constexpr const auto timeout = std::chrono::seconds(2);
     auto start = std::chrono::system_clock::now();
+    bool wasStuck = false;
     do {
         _engine->getAlicaClock().sleep(alica::AlicaTime::milliseconds(ALICA_LOOP_TIME_ESTIMATE));
         if (std::chrono::system_clock::now() > start + timeout) {
-            throw std::runtime_error("Got stuck trying to step engine");
+            if (!wasStuck) {
+                // avoid spamming by logging only the first step we're stuck
+                Logging::logWarn(LOGNAME) << "Got stuck trying to step engine for too long";
+                wasStuck = true;
+            }
+        } else {
```

Comment:
Since this is within a single step, I expect setting this to false doesn't impact anything as we will leave the function scope soon after anyway